third_party/libtock-drivers/src/nfc.rs

```diff
@@ -0,0 +1,144 @@
+use crate::util;
+use core::cell::Cell;
+use libtock_core::{callback, syscalls};
+
+const DRIVER_NUMBER: usize = 0x30003;
+
+mod command_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const EMULATE: usize = 3;
+    pub const CONFIGURE: usize = 4;
+    pub const FRAMEDELAYMAX: usize = 5;
+}
+
+mod subscribe_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const SELECT: usize = 3;
+}
+
+mod allow_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+}
+
+pub struct NfcTag {}
+
+impl NfcTag {
+    pub fn enable_emulation() {
+        NfcTag::emulate(true);
+    }
+
+    pub fn disable_emulation() {
+        NfcTag::emulate(false);
+    }
+
+    pub fn emulate(enabled: bool) -> bool {
+        let result_code =
+            syscalls::command(DRIVER_NUMBER, command_nr::EMULATE, enabled as usize, 0);
+        if result_code.is_err() {
+            return false;
+        }
+
+        true
+    }
+
+    /// Subscribe to the tag being SELECTED callback.
+    pub fn selected() -> bool {
+        let is_selected = Cell::new(false);
+        let mut is_selected_alarm = || is_selected.set(true);
+        let subscription = syscalls::subscribe::<callback::Identity0Consumer, _>(
+            DRIVER_NUMBER,
+            subscribe_nr::SELECT,
+            &mut is_selected_alarm,
+        );
+        if subscription.is_err() {
+            return false;
+        }
+
+        util::yieldk_for(|| is_selected.get());
+        true
+    }
+
+    /// Configure the tag type command.
+    pub fn configure(tag_type: u8) -> bool {
+        let result_code =
+            syscalls::command(DRIVER_NUMBER, command_nr::CONFIGURE, tag_type as usize, 0);
+        if result_code.is_err() {
+            return false;
+        }
+
+        true
```

Comment:
See above for `is_ok()` or switching to returning error codes. Also, you don't need a newline here in my opinion.

Same goes for functions below.